### PR TITLE
validate DPG configuration

### LINKED ISSUE #1602: container-network-gateway/dns/ip-range allow either DPG name or network name
As discussed with @emlin, currently, when you specify `--container-network-gateway/dns/ip-range`, `vic-machine create` allows you to specify either the distributed port group name or the descriptive network name that you specify in `--container-network`. If you use the descriptive network name, deployment succeeds and containers appear to be using the correct network, but creating a container that uses that network will result in errors. Only the DPG name works correctly.

We should either warn the user if the wrong parameter is provided, or allow both possibilities.

### PR DESCRIPTION
**Pull request by emlin**
Fixes #1602 
Warning user if container network gateway, ip range or dns configured, but corresponding network is not set.

### CODE REVIEW

**Comment by emlin** [comment]
```
./bin/vic-machine-linux create --target=root:ca\$hc0w@10.17.109.69/dc1 --image-store=vsanDatastore --bridge-network vch-test --container-network InternalNetwork:internal --container-network-gateway internal:1.1.1.1/16 --container-network-ip-range a_network:172.16.0.10-20 --container-network-dns a_network:8.8.8.8
WARN[2016-08-22T09:21:59-05:00] Following container network gateway is set, but no container network for it. Please check your --container-network and --container-network-gateway 
WARN[2016-08-22T09:21:59-05:00] 	internal:1.1.1.1/16                         
WARN[2016-08-22T09:21:59-05:00] Following container network ip range is set, but no container network for it. Please check your --container-network and --container-network-ip-range 
WARN[2016-08-22T09:21:59-05:00] 	a_network:[{172.16.0.10 172.16.0.20}]       
WARN[2016-08-22T09:21:59-05:00] Following container network dns is set, but no container network for it. Please check your --container-network and --container-network-ip-range 
WARN[2016-08-22T09:21:59-05:00] 	a_network:[8.8.8.8]                         
INFO[2016-08-22T09:21:59-05:00] ### Installing VCH ####                      
```

**Comment by emlin** [comment]
Improved error message as Andrew's comments. 
```
./bin/vic-machine-linux create --target=root:ca\$hc0w@10.17.109.69/dc1 --image-store=vsanDatastore --bridge-network vch-test --container-network InternalNetwork:internal --container-network-gateway internal:1.1.1.1/16 --container-network-ip-range a_network:172.16.0.10-20 --container-network-dns a_network:8.8.8.8  --container-network-gateway internal1:1.1.1.2/16
ERRO[2016-08-22T09:59:02-05:00] Following container network gateway is set, but CONTAINER-NETWORK not found. Please check your --container-network and --container-network-gateway 
ERRO[2016-08-22T09:59:02-05:00] 	internal:1.1.1.1/16, "internal" should be vsphere network name 
ERRO[2016-08-22T09:59:02-05:00] 	internal1:1.1.1.2/16, "internal1" should be vsphere network name 
ERRO[2016-08-22T09:59:02-05:00] Following container network ip range is set, but CONTAINER-NETWORK not found. Please check your --container-network and --container-network-ip-range 
ERRO[2016-08-22T09:59:02-05:00] 	a_network:[{172.16.0.10 172.16.0.20}], "a_network" should be vsphere network name 
ERRO[2016-08-22T09:59:02-05:00] Following container network dns is set, but no container network for it. Please check your --container-network and --container-network-ip-range 
ERRO[2016-08-22T09:59:02-05:00] 	a_network:[8.8.8.8], "a_network" should be vsphere network name 
Inconsistent container network configuration.
```

**Comment by andrewtchin** [comment]
lgtm

File: cmd/vic-machine/create/create.go [hunk 1]
```diff
@@ -379,8 +379,37 @@ func (c *Create) processContainerNetworks() error {
...
 		c.MappedNetworksDNS[vicnet] = dns[vnet]
 
+		delete(gws, vnet)
+		delete(pools, vnet)
+		delete(dns, vnet)
 	}
 
+	var hasError bool
+	if len(gws) > 0 {
+		log.Errorf("Following container network gateway is set, but CONTAINER-NETWORK not found. Please check your --container-network and --container-network-gateway")
```
Comment (stuclem) [inline cmd/vic-machine/create/create.go]: Should be: "The following container network gateway is set, but CONTAINER-NETWORK cannot be found. Please check the --container-network and --container-network-gateway settings"

File: cmd/vic-machine/create/create.go [hunk 2]
```diff
@@ -379,8 +379,37 @@ func (c *Create) processContainerNetworks() error {
...
+		for key, value := range gws {
+			mask, _ := value.Mask.Size()
+			log.Errorf("\t%s:%s/%d, %q should be vsphere network name", key, value.IP, mask, key)
+		}
+		hasError = true
+	}
+	if len(pools) > 0 {
+		log.Errorf("Following container network ip range is set, but CONTAINER-NETWORK not found. Please check your --container-network and --container-network-ip-range")
```
Comment (stuclem) [inline cmd/vic-machine/create/create.go]: Should be: "The following container network IP range is set, but CONTAINER-NETWORK cannot be not found. Please check the --container-network and --container-network-ip-range settings"

File: cmd/vic-machine/create/create.go [hunk 3]
```diff
@@ -379,8 +379,37 @@ func (c *Create) processContainerNetworks() error {
...
+		for key, value := range pools {
+			log.Errorf("\t%s:%s, %q should be vsphere network name", key, value, key)
```
Comment (stuclem) [inline cmd/vic-machine/create/create.go]: Should be: " should be vSphere network name"

File: cmd/vic-machine/create/create.go [hunk 4]
```diff
@@ -379,8 +379,37 @@ func (c *Create) processContainerNetworks() error {
...
+		}
+		hasError = true
+	}
+	if len(dns) > 0 {
+		log.Errorf("Following container network dns is set, but no container network for it. Please check your --container-network and --container-network-ip-range")
```
Comment (stuclem) [inline cmd/vic-machine/create/create.go]: Should be: "The following container network DNS is set, but there is no container network for it. Please check the --container-network and --container-network-ip-range settings."

File: cmd/vic-machine/create/create.go [hunk 5]
```diff
@@ -401,8 +401,37 @@ func (c *Create) processContainerNetworks() error {
...
 		c.MappedNetworksDNS[vicnet] = dns[vnet]
 
+		delete(gws, vnet)
+		delete(pools, vnet)
+		delete(dns, vnet)
 	}
 
+	var hasError bool
+	if len(gws) > 0 {
+		log.Errorf("The following container network gateway is set, but CONTAINER-NETWORK cannot be found. Please check the --container-network and --container-network-gateway settings")
```
Comment (jzt) [inline cmd/vic-machine/create/create.go]: Minor, but you could avoid some DRY by doing something like this:

```go
errorString := "The following container network gateway is set, but CONTAINER-NETWORK cannot be found. Please check the --container-network and %s settings"
log.Errorf(errorString, "--container-network-gateway")
log.Errorf(errorString, "--container-network-ip-range")
log.Errorf(errorString, "--container-network-dns")
```

**Comment by jzt** [comment]
LGTM 👍

File: cmd/vic-machine/create/create.go (hunk 5 above)
Comment (emlin) [inline cmd/vic-machine/create/create.go]: good point

**Comment by emlin** [comment]
♻️

**Comment by emlin** [comment]
♻️

**Pull request merged**